### PR TITLE
Improvements of M1M3SIMGUI

### PR DESCRIPTION
**Pull request by pkubanek**
- Ignore comments in CSV, report filename with parser error
- Config directory can be specified as argument

### CODE REVIEW

File: python/lsst/ts/criopy/gui/sal/application.py [hunk 1]
```diff
@@ -110,10 +132,17 @@ def add_comm(
         """
         self._comms_args.append(SplashScreen.CommArgs(name, manual, kwargs))
 
+    def process_command_line(self) -> None:
+        """Called to process command line arguments. Shall be overridden in
+        children, if custom command line processing is desired. Use self.parser
+        to access command line parser."""
+        pass
```
Comment (teweitsai) [inline python/lsst/ts/criopy/gui/sal/application.py:139]: How about to `raise` the `NotImplementedError` instead?

File: python/lsst/ts/criopy/gui/sal/splash_screen.py [hunk 2]
```diff
@@ -86,8 +87,18 @@ def _checkStarted(self) -> None:
         if len(self.comms) == 0 and len(self.comms_args) > 0:
             # make sure the loop was started
             assert asyncio.get_event_loop().is_running()
-            for args in self.comms_args:
-                self.comms.append(create(args.name, manual=args.manual, **args.kwargs))
+            try:
+                for args in self.comms_args:
```
Comment (teweitsai) [inline python/lsst/ts/criopy/gui/sal/splash_screen.py:91]: `for arg in self.comms_args:`

File: python/lsst/ts/criopy/m1m3simgui.py [hunk 3]
```diff
@@ -90,23 +91,36 @@ def reload_config(self) -> None:
...
             return
+        self.load_config(config_dir)
+
+    def load_config(self, config_dir: str) -> None:
+        """Load new configuration. Can be called from application to load
+        configuration.
+
+        Parameters
+        ----------
+        config_dir L `str`
```
Comment (teweitsai) [inline python/lsst/ts/criopy/m1m3simgui.py:109]: """
config_dir : `str`
"""

**Review by teweitsai — APPROVED** [review]
Looks good.

File: python/lsst/ts/criopy/gui/sal/application.py (hunk 1 above)
Comment (pkubanek) [reply python/lsst/ts/criopy/gui/sal/application.py:139]: It's perfectly legal to not process custom command line arguments if there aren't any custom command line arguments. This is a placeholder for children to override.

File: python/lsst/ts/criopy/m1m3simgui.py (hunk 3 above)
Comment (pkubanek) [reply python/lsst/ts/criopy/m1m3simgui.py:109]: Fixed

**Pull request merged**